src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/Kavita.kt

```diff
@@ -382,6 +384,39 @@ class Kavita(private val suffix: String = "") : ConfigurableSource, UnmeteredSou
             statements = mutableListOf(),
         )
 
+        // Handle text query first
+        if (query.isNotBlank()) {
```

Comment:
For the future, you might want to wrap these into methods that either return something or null or just do the modification. This will keep the code easier to understand.